### PR TITLE
Add option to disable lockfile generation

### LINKED ISSUE #1094: Provide ability to opt-out of lockfile generation
# Overview
Manifest support was added in CLI v5.1.0 to allow analyzing manifest files by generating lockfiles on-demand. This issue is to provide (more?) mechanisms for opting out of this behavior.

## Additional Context

If there are already means to opt-out of the lockfile generation (e.g., specifying explicit entries in the `.phylum_project` file), they should be documented more clearly. For example, that could be in the form of an update to a current page on the https://docs.phylum.io/ site or maybe a new page (e.g., KB article for "manifests vs. lockfiles" or perhaps a new Troubleshooting/Tips/FAQ page).

## Acceptance Criteria
- [x] Automatic lockfile generation is retained as the default behavior
- [x] Opting out of automatic lockfile generation is possible for use cases that do not involve the `.phylum_project` file
- [x] Documentation is created/updated to be more clear about the options for opt-out

### PR DESCRIPTION
**Pull request by cd-work**
Closes #1094

### CODE REVIEW

**Review by maxrake — CHANGES_REQUESTED** [review]
First pass quick review:
* CHANGELOG entry needed
* Docs need to be regenerated
* There is a deno check failure in CI
* Add more detail to the [lockfile generation](https://docs.phylum.io/docs/lockfile_generation) page for explaining the options for opt-out

File: cli/src/commands/parse.rs [hunk 1]
```diff
@@ -98,8 +103,8 @@ pub fn parse_lockfile(
         }
     }
 
-    // If the path is neither a valid manifest nor lockfile, we abort.
-    if !parser.is_path_manifest(&path) {
+    // Abort if generation is disabled or path is neither lockfile nor manifest.
+    if !generate_lockfiles || !parser.is_path_manifest(&path) {
```
Comment (maxrake) [inline cli/src/commands/parse.rs:107]: There is a problem with the logic in this block. The `unreachable` macro is triggered with the changes from this PR. Here is an example from [a library project](https://github.com/deepcausality-rs/deep_causality) where there are only manifests:

```
../deep_causality  9  19 on  main [?] via 🦀 v1.73.0
❯ /Users/maxrake/dev/phylum/localdev/cli/target/debug/phylum --version
phylum v5.8.1-4-ge70c5c1

../deep_causality  9  19 on  main [?] via 🦀 v1.73.0
❯ /Users/maxrake/dev/phylum/localdev/cli/target/debug/phylum parse --no-generation ./Cargo.toml
thread 'main' panicked at cli/src/commands/parse.rs:112:21:
internal error: entered unreachable code: neither lockfile nor manifest
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Review by maxrake — CHANGES_REQUESTED** [review]
The code was reviewed. The doc updates and changelog entries look good. Local testing revealed an issue.

File: cli/src/commands/parse.rs (hunk 1 above)
Comment (cd-work) [reply cli/src/commands/parse.rs:107]: Great catch, thanks.

**Review by maxrake — DISMISSED** [review]
LGTM

**Review by maxrake — APPROVED** [review]
Code reviewed and tested locally. LGTM.

**Pull request merged**